src/app/components/date-range-dropdown/date-range-dropdown.component.ts

```diff
@@ -0,0 +1,73 @@
+import { ChangeDetectionStrategy, Component, forwardRef, inject, OnDestroy, OnInit } from '@angular/core';
+import { ControlValueAccessor, FormControl, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { Subject, takeUntil } from 'rxjs';
+import { DateRange, DateRangeDropdown, DATE_RANGE_DROPDOWN_DATA } from './../../models';
+import { DateService } from './../../services';
+
+@Component({
+	selector: 'app-date-range-dropdown',
+	templateUrl: './date-range-dropdown.component.html',
+	styleUrls: ['./date-range-dropdown.component.scss'],
+	changeDetection: ChangeDetectionStrategy.OnPush,
+	providers: [
+		{
+			provide: NG_VALUE_ACCESSOR,
+			useExisting: forwardRef(() => DateRangeDropdownComponent),
+			multi: true,
+		},
+	],
+})
+export class DateRangeDropdownComponent implements OnInit, OnDestroy, ControlValueAccessor {
+	// inner FormGroup used to manage ControlValueAccessor value
+	readonly dateRangeDropdownFormControl = new FormControl<DateRangeDropdown | null>(null, {});
+	private readonly destroyer$ = new Subject<void>();
+	private readonly dateService = inject(DateService);
+
+	DATE_RANGE_DROPDOWN_DATA = DATE_RANGE_DROPDOWN_DATA;
+
+	// onChange callback that will be overridden using `registerOnChange`
+	onChange: (dateRange?: DateRange | null) => void = () => {
+		/** empty */
+	};
+
+	// onTouched callback that will be overridden using `registerOnTouched`
+	onTouched = () => {
+		/** empty */
+	};
+
+	ngOnInit(): void {
+		this.dateRangeDropdownFormControl.valueChanges.pipe(takeUntil(this.destroyer$)).subscribe((value) => {
+			if (value) {
+				const dateRange = this.dateService.getDateRangeByDropdown(value);
+				this.onChange(dateRange);
+			}
+		});
+	}
+
+	/**
+	 * Set Component's ControlValueAccessor value
+	 */
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
This shouldn't be needed

Reply:
it is needed, because right now we are not doing anything with `dateRange`